projects/plugins/jetpack/class.json-api-endpoints.php

```diff
@@ -1521,6 +1523,26 @@ public function get_author( $author, $show_email_and_ip = false ) {
 			$author['site_visible'] = $site_visible;
 		}
 
+		// Only include WordPress.com user data when author_wpcom_data is enabled.
+		$args              = $this->query_args();
+		$author_wpcom_data = isset( $args['author_wpcom_data'] ) && $args['author_wpcom_data'];
+
+		if ( ! empty( $id ) && $author_wpcom_data ) {
+			// If this is a Jetpack site, use the connection manager to get the user data.
+			if ( ! defined( 'IS_WPCOM' ) || ! IS_WPCOM ) {
+				$connection_manager = new \Automattic\Jetpack\Connection\Manager();
+				$wpcom_user_data    = $connection_manager->get_connected_user_data( $id );
+				if ( $wpcom_user_data && isset( $wpcom_user_data['ID'] ) ) {
+					$author['wpcom_id']    = (int) $wpcom_user_data['ID'];
+					$author['wpcom_login'] = isset( $wpcom_user_data['login'] ) ? $wpcom_user_data['login'] : '';
+				}
+			} else {
+				$user                  = get_user_by( 'id', $id );
+				$author['wpcom_id']    = isset( $user->ID ) ? (int) $user->ID : null;
+				$author['wpcom_login'] = isset( $user->user_login ) ? $user->user_login : '';
```

Comment:
Maybe we can simplify this into the below. That's assuming, like above, that we don't need to do any kind of escaping there though.

```suggestion
				$author['wpcom_login'] = $user->user_login ?? '';
```